infra/types/DeploymentSettings.bicep

```diff
@@ -22,6 +22,9 @@ type DeploymentSettings = {
   @description('The ID of the principal that is being used to deploy resources.')
   principalId: string
 
+  @description('The name of the principal that is being used to deploy resources.')
+  principalName: string
```

Comment:
This might be the time to switch over to includes for types.  See https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-import 

Since you are touching DeploymentSettings everywhere, mark the type here with @export(), then swap out the type definitions everywhere else with an import statement.

Reply:
yup, although I believe we're using an older version of bicep that doesn't suppor this. I plan on updating this so we don't have to deal with this in the future